pysph/tools/interpolator.py

```diff
@@ -209,19 +313,30 @@ def interpolate(self, prop, gradient=False):
         prop: str
             The name of the property to interpolate.
 
-        gradient: bool
-            Evaluate gradient and not function.
+        comp: int
+            The component of the gradient required
 
         Returns
         -------
         A numpy array suitably shaped with the property interpolated.
         """
+        assert isinstance(comp, int), 'Error: only interger value is allowed'
         for array in self.particle_arrays:
             data = array.get(prop, only_real_particles=False)
             array.get('temp_prop', only_real_particles=False)[:] = data
 
         self.func_eval.compute(0.0, 0.1)  # These are junk arguments.
-        result = self.pa.prop.copy()
+        if comp and self.use_shepard:
+            print("Error: set use_shepard=False to evaluate gradients")
```

Comment:
Raise an exception here, you could for example do `raise RuntimeError('Set use_shepard=False ...')`, otherwise someone may choose to ignore the warning and accept the result as correct.

Reply:
done